### ISSUE TITLE
Questions about gpu

### ISSUE THREAD

**Issue by iceyyyoung** [Svardfox/LaViT#3]
Hi, thanks for your great work! 

Could you please share what GPU you used and the minimum VRAM required to run this inference/training? Also, I was wondering when the LaViT-15k might be available. Thanks!

**Comment by Svardfox** [comment]
Hi @iceyyyoung, thank you for your interest in our work!

Regarding your questions:

GPU Usage: For training LaViT-3B, we only used a single Nvidia H20. However, for the dataset construction phase, we utilized 4x H20 GPUs as the process relied on eager mode.

Dataset Availability: We are currently finalizing the organization of the data and will open-source LaViT-15k as soon as possible.
Please stay tuned for updates!

**Comment by iceyyyoung** [comment]
Thank you for your reply. I have one more question.

In my experiments, according to your descriptions:

```
--force_lvr: Force append <lvr> tokens to the prompt (test if model uses bottleneck tokens)
--mask_lvr: Mask <lvr> tokens in attention mask (test model's reliance on them)
```

after enabling these two options, I observed that the evaluation results of LaVit-3B on MMVP remain exactly the same as those obtained without masking the `lvr` tokens.

I am wondering whether this behavior is expected, or if there might be something incorrect in my usage or implementation.

**Comment by Svardfox** [comment]
Thanks for reporting this. The unchanged MMVP results are expected with the released LaViT-3B checkpoint, not a bug.

- `--force_lvr` just appends 4 generic `<lvr>` tokens; `--mask_lvr` only zeroes their attention mask. Inference logic doesn’t change.
- The bottleneck mask is only applied in training stage 1 when `self.training == True` inside `LaViTQwen2VL.forward`, so eval mode never triggers it.
- Training used numbered tokens `<lvr1>…<lvr4>`, while the eval script appends plain `<lvr>` tokens, which aren’t trained and carry no useful signal.

If you want to test reliance on the bottleneck, evaluate a stage-1 checkpoint, remove the `and self.training` guard so the bottleneck mask is active during inference, and append `<lvr1><lvr2><lvr3><lvr4>` in the prompt. Otherwise, with the joint model, these flags are effectively no-ops and won’t change outputs.

**Comment by iceyyyoung** [comment]
Thank you very much for your reply. However, it seems that the stage-1 checkpoint has not been released yet.

Besides, I would like to ask about the implementation of Table 3: “w/o Latent Tokens – Masking Latent Tokens at Inference” in your paper. My understanding is that this setting tests the final model’s reliance on the four latent thinking tokens `<lvr1>…<lvr4>` by masking them during inference so that the model cannot attend to these tokens when generating the answer. Is this correct?

If so, could you please clarify how this masking is implemented in practice during inference?

Thank you again for your time and help!

**Comment by YingYellow** [comment]
How much memory is consumed when training on an H20 GPU?

**Comment by Svardfox** [comment]
Hi, thanks for the thoughtful questions!

You are correct on both points:

1) Stage-1 checkpoint  
At the moment, we have not released a separate stage-1 checkpoint yet.

2) Table 3 (“w/o Latent Tokens – Masking Latent Tokens at Inference”)  
Your understanding is correct: this setting tests whether the final model relies on latent tokens by preventing the model from attending to them during generation.

In our inference code, this is implemented by zeroing the `attention_mask` at latent-token positions (rather than deleting tokens from the sequence). Concretely:
- we locate latent-token IDs in `input_ids`;
- set their corresponding entries in `attention_mask` to 0;
- then run normal `generate()`.

With this implementation, latent tokens are still present in the sequence, but they are masked out as attention targets during decoding, so generated answer tokens cannot attend to them.

Practical note: if there are no latent tokens in the input, masking has no effect. To explicitly test dependence, we use the “force latent tokens + mask latent tokens” setting together.

**Comment by Svardfox** [comment]
Hi, thanks for your question!

We trained LaViT-3B on a single Nvidia **PRO 6000 (96GB)** without flash attention.

With the default settings in our scripts:
- `per_device_train_batch_size=1`
- `gradient_accumulation_steps=16`
- `gradient_checkpointing=True`
- `bf16=True`

The training fits comfortably within the 96GB memory.

**Comment by iceyyyoung** [comment]
Hi,
I’ve encountered another specific phenomenon while experimenting and would appreciate your insights.

In SFT stage of your code, I notice that all labels begin with the` <lvr> `token (e.g., `<lvr><lvr><lvr><lvr> A`). However, during evaluation, the model's output depends heavily on the prompt: when the prompt includes a strict instruction like _"Answer with the option letter from the given choices directly"__(MMVP benchmark),_ the model often bypasses the `<lvr>` tokens and outputs "A" directly; yet, if I remove such formatting constraints, it correctly prefixes its response with `<lvr> `as it was trained to do.

Have you encountered similar situations where the model reverts to its original behavior under specific prompts? From your experience, what is the underlying mechanism behind this, and are there any recommended strategies to resolve it?

Thank you very much for taking the time to read this and for sharing your expertise. Your response would be incredibly helpful to my work!

Best regards
